.circleci/parse_yaml.py

```diff
@@ -0,0 +1,269 @@
+
+"""Reading and parsing through the contributor's yaml file.
+(c) YoSon Park
+
+This is the default uploading skim for single and bulk files 
+using zenodo. For PR-based file uploader, check .circleci/deploy.sh 
+and .circleci/process_submitted_data.py
+"""
+
+
+import sys
+import yaml
+import pandas
+import pathlib
+import logging
+
+
+CONFIG_FILE_EXTENSION = ('yml', 'yaml')
+# transparent compressions recognized by tarfile 'r:*' are:
+# gzip, bz2, and lzma (xz)
+COMPRESSION_EXTENSION = ('gz', 'bz2', 'xz')
+BULK_EXTENSION = ('tgz', 'tar.gz', 'tar.bz2', 'tar.xz')
+FILE_EXTENSION = ('txt', 'tsv')
+
+logging.basicConfig(level='INFO')
+
+
+def check_yaml_before_commit():
+    """A git-status-dependent function used when locally applying 
+    parse_yaml.py. It searches for a new or modified yml/yaml file and 
+    returns its pathlib path.
+    """
+    import subprocess
+
+    status_check = 'git status --porcelain'
+    jobsplit = subprocess.check_output(
+        status_check,
+        stderr=subprocess.STDOUT,
+        shell=True,
+        universal_newlines=True
+    ).replace('\n', '').split('?? ')
+
+    for job in jobsplit:
+        if job.endswith(CONFIG_FILE_EXTENSION):
+            yaml_file = pathlib.Path('.cimr-d/' + job.split('/')[-1])
+
+    return yaml_file
+
+
+def check_yaml_in_ci():
+    """A git-status-dependent function used during ci processing. It 
+    searches for a new or modified yml/yaml file from a new pr"""
+    import subprocess
+
+    status_check = 'git diff origin --name-only'
+    jobsplit = subprocess.check_output(
+        status_check,
+        stderr=subprocess.STDOUT,
+        shell=True,
+        universal_newlines=True
+    ).replace('\n', '').split('?? ')
```

Comment:
Note that if you plan to use this function, this line should be changed into:
```python
).split('\n')
```
because the output of `git diff origin --name-only` command doesn't have the leading `?? ` characters.